examples/sgemm_interop/sycl_sgemm.cpp

```diff
@@ -104,12 +86,14 @@ int main() {
       auto d_C = b_C.get_access<sycl::access::mode::write>(h);
 
       h.host_task([=](sycl::interop_handle ih) {
-        cuCtxSetCurrent(ih.get_native_context<backend::ext_oneapi_cuda>());
```

Comment:
Is this change related to the change in https://github.com/intel/llvm/pull/8197 to use the primary context?

Reply:
We probably do need to keep these, were they causing issues?

Reply:
If we are only using the Primary Context , then is this really needed? If the SYCL context is detached from the actual CUDA context, and we follow the CUDA Runtime approach, then the developer has no notion of CUDA contexts at all.  
`cuCtxtSetCurrent` is a CUDA Driver API, not Runtime. Shouldn't the host_task implementation in this case just set the current context to the primary one?

Reply:
From https://github.com/intel/llvm/pull/8197 description (pretty good btw!):

```quote
- Likely more efficient in the general case, with less opportunities to accidentally cause costly CUDA context switches.
```
But if the explicit call to set current is needed then its easily to miss it, or to change it without needing it, since there is no guarantee the host_task will execute on the user calling thread for the queue submission anyway.

```quote
- Easier and likely more efficient interactions with CUDA runtime applications.
```

The usage of cuBLAS that we are representing here is that of a CUDA runtime application, so if its possible we should avoid the Context if we can, the code would look nicer from an user perspective.

Reply:
> Shouldn't the host_task implementation in this case just set the current context to the primary one?

That's a good idea, we should probably do this, but I don't believe we're doing it currently, even when using the primary context it still needs to be set on the thread, that's why I think technically we still need the `cuCtxSetCurrent` right now, until we tweak the host task implementation to set it up automatically.

Reply:
Changing host task turned out to be a rabbit hole that requires PI changes and maybe future UR changes. I am working on those separately, and adding here manually the context change.